python/beeai_framework/adapters/ollama/backend/chat.py

```diff
@@ -28,6 +28,9 @@ def provider_id(self) -> ProviderName:
         return "ollama"
 
     def __init__(self, model_id: str | None = None, settings: dict | None = None) -> None:
+        if not hasattr(settings, "base_url") and "OLLAMA_BASE_URL" in os.environ:
+            settings["base_url"] = os.getenv("OLLAMA_BASE_URL")
```

Comment:
if `settings` is None, i believe this (`settings["base_url"]`) would fail

Reply:
You're right, although I'm not sure settings can be None by this point in the code.  The following test case (with no settings specified) should test for this:

```python
    # Ollama with Granite model and base_url specified in env var
    os.environ["OLLAMA_BASE_URL"] = "http://somewhere-else:12345"
    ollama_chat_model = ChatModel.from_name("ollama:granite3.1-dense:8b")
    assert isinstance(ollama_chat_model, OllamaChatModel)
    assert ollama_chat_model.settings["base_url"] == "http://somewhere-else:12345"
```

That said, I think it's better to be safer so I'll add something to check.

Reply:
Resolved in [a74c44c](https://github.com/i-am-bee/beeai-framework/pull/389/commits/a74c44c561a0892f75b2211aa2417e1918d414d8)